src/service/users.js

```diff
@@ -15,16 +15,21 @@ export const verifyUser = async (userId) => {
   }
 };
 
-export const demoVerifyUser = async (userId) => {
+export const demoVerifyUser = async (user_id, password) => {
   try {
-    const resApi = await api.post('/users/demo-verify-user', {
-      user_id: userId
+    const resApi = await api.post('/users/demo-verify-user-v2', {
+      user_id,
+      password
     });
 
     return resApi.data;
   } catch (error) {
     crashlytics().recordError(new Error(error));
-    return error.response.data;
+    if (error?.response?.data?.status?.toLowerCase() === 'password is invalid') {
+      throw new Error('Password is invalid');
+    }
+
+    return error?.response?.data;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the `demoVerifyUser` function are generally good. The addition of password verification is a positive step towards improving security. However, there's an issue with error handling. The check for 'password is invalid' is case-sensitive which might lead to problems if the server returns the message in a different case. It would be better to make this comparison case-insensitive.

```diff
-    if (error?.response?.data?.status?.toLowerCase() === 'password is invalid') {
+    if (error?.response?.data?.status?.toLowerCase() === 'password is invalid'.toLowerCase()) {
       throw new Error('Password is invalid');
     }
```
Also, it's not clear what type of object `error.response.data.status` is. If it's not a string, calling `toLowerCase()` on it will cause a runtime error. You should add a type check before calling `toLowerCase()`.

```diff
-    if (error?.response?.data?.status?.toLowerCase() === 'password is invalid'.toLowerCase()) {
+    if (typeof error?.response?.data?.status === 'string' && error?.response?.data?.status?.toLowerCase() === 'password is invalid'.toLowerCase()) {
       throw new Error('Password is invalid');
     }
```

Lastly, you're returning `error?.response?.data` when an error occurs. This could potentially leak sensitive information. It would be better to return a generic error message or create a custom error object that only includes necessary information.

```diff
-    return error?.response?.data;
+    return { message: 'An error occurred while verifying the user.' };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
no problem, since we have optional chaining and the text lowercased is response from the server.